capstone/src/main/webapp/youtubeScript.js

```diff
@@ -21,39 +21,61 @@ const ATTRIBUTES_BY_LANGUAGE = {
   'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
 };
 
+/** Category names are mapped to youtube category numbers */
+const YOUTUBE_CATEGORIES = {
+  'Autos&Vehicles': 2,
+  'Comedy': 23,
+  'Entertainment': 24,
+  'Film&Animation': 1,
+  'Gaming': 20,
+  'How-to&Style' : 26,
+  'Music': 10,
+  'Pets&Animals': 15,
+  'Science&Technology': 28,
+  'Sports' : 17,
+};
+
 /** Calls youtube servlet and passes output to perspctive */
 async function callYoutube() {
+  document.getElementById('search-type').innerHTML = "";
   const channelId = document.getElementById('channelIdForAnalysis').value.replace(/ /g, '');
   if (!channelId) {
     return;
   }
-  var response;
+  /** Checks if input is a category, if so directs input to be handled by get trending*/
+  if (YOUTUBE_CATEGORIES[channelId] != undefined) {
+    document.getElementById('search-type').innerHTML = "Category Search";
+    getTrending(YOUTUBE_CATEGORIES[channelId]);
+    return;
+  }
   /** Checks if input follows channel ID format, if not attempts to convert it to channel ID*/
+  var response;
+  var responseJson;
```

Comment:
nit: The Google Style Guide suggests we always use `let` instead of `var`, so I would change these, as well as the declarations in lines 108 and 109.

Reply:
addressed in "change var to let"